### ISSUE TITLE
import-blacklist not working for named imports

### ISSUE THREAD

**Issue by Stupar92** [palantir/tslint#4400]
### Bug Report
Import blacklist is not working on named imports.

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.1.6
- __Running TSLint via__: (pick one) CLI

#### TypeScript code being linted

```ts
import { action } from "mobx";
```

with `tslint.json` configuration:

```json
        "import-blacklist": [
            true,
            {
                "mobx": [
                    "action"
                ]
            }
        ]
```

#### Actual behavior
Ts lint passes with no issues.
#### Expected behavior
It should fail since using `action` from mobx is blacklisted.

**Comment by JoshuaKGoldberg** [comment]
@Stupar92 I get the error locally. What command are you using to run the CLI?

**Comment by JoshuaKGoldberg** [comment]
Closing for inactivity. @Stupar92 please do comment again if you're seeing the bad behavior!

**Comment by Mr1upMachine** [comment]
@JoshuaKGoldberg I was able to reproduce the issue above. I am using a base Angular CLI project, which uses typescript `~3.2.2` and tslint `~5.11.0`. I attempted it with installing lodash and blacklisting `isFunction` like this:
```
"import-blacklist": [
  true,
  "rxjs/Rx",
  {
    "lodash": [
      "isFunction"
    ]
  }
],
```
I will try and see if I can reproduce the issue in a different context, however a fresh Angular CLI application should work.

**Comment by christopheberth** [comment]
@JoshuaKGoldberg same issue over here. Rule was added but tslint is not marking any issue.
Could you re-open this issue?

@Mr1upMachine did you find a solution to fix?

Current configuration:
`@angular/core: 5.2.7`
`@angular/cli: 1.7.4`
`@angular/compiler-cli: 5.2.7`
`Typescript 3.2.2.`
`Tslint 5.9.1`

**Comment by Mr1upMachine** [comment]
@christopheberth I have not yet no.

**Comment by bopfer** [comment]
Same issue here my React/Typescript app.  This is not working:

```
"import-blacklist": [
  true,
  {
    "styled-components": [
      {
        "name": "styled",
        "message": "import from styled-components/macro instead"
      }
    ]
  }
]
```

typescript: 3.3.3333
tslint: 5.13.0

**Comment by JoshuaKGoldberg** [comment]
I just tried the OP's sample code with `mobx` and `action` using TSLint 5.13.0 and correctly got a TSLint complaint.

```json
{
  "rules": {
    "import-blacklist": [
      true,
      {
        "mobx": [
          "action"
        ]
      }
    ]
  }
}
```

```ts
// index.ts
import { action } from "mobx";
```

@bopfer @Mr1upMachine @christopheberth please share:

* Your `tslint.json`
* Your TypeScript and TSLint versions _(if not typescript@3.3 and tslint@5.13, please try again with those)_ 
* How you're running TSLint _(command line? particular IDE extension? ...?)_
* Minimal sample code from a TypeScript file that should get a lint failure but doesn't

Without all those, it's rather difficult to repro what's going wrong.

**Comment by bopfer** [comment]
@JoshuaKGoldberg, I have create a bare minimum repo to show the issue I am having.  This repo is using the exact `tslint.json` file that I am using in my real app:

https://github.com/bopfer/import-blacklist-testing

I would like this to be blacklisted:

```javascript
import styled from 'styled-components';
```

In favor of this:

```javascript
import styled from 'styled-components/macro';
```

But, in `src/index.tsx`, VSCode and `yarn lint` will not report it.

It's very possible I am just plain doing something wrong, but I am not sure what that would be.

**Comment by JoshuaKGoldberg** [comment]
Super, thanks @bopfer! I'll take a look.

**Comment by villelahdenvuo** [comment]
Also hitting this issue. I tried to upgrade dependencies, but it didn't help.

```
tslint: 5.15.0
typescript: 3.3.3333
typescript-tslint-plugin: 0.3.1
```

**Comment by villelahdenvuo** [comment]
Actually this is still a problem, since it does not work in jsRules for some reason. I don't see why it shouldn't work for regular ES import statements.

**Comment by JoshuaKGoldberg** [comment]
I never did take a look 😞... but thanks @villelahdenvuo! It sounds like the issue is with the rule's documentation and/or config parsing then?

**Comment by villelahdenvuo** [comment]
I think the documentation is correct, I just saw the wrong example in some stackoverflow post. And my initial issue is with the VS Code/typescript plug-in because it's not showing lint errors for js files even though it is configured to do so. When I run it via the cli it works fine.

**Comment by josh-lind** [comment]
So is there any way to get this working with the VS Code/typescript plug-in. No one in my company actually uses ng lint (just linting on save) so it would be nice to have a fix for this. I've tried both of these rules 
`"import-blacklist": [
      true,
      [
        "^primeng.primeng$",
        "^rxjs.Rx$"
      ]
    ],
    "import-blacklist": [
      true,
      "rxjs/Rx",
      "primeng/primeng"
    ],`
and neither throws an error when writing `import { MessageModule } from 'primeng/primeng';`

**Comment by YoussefTaghlabi** [comment]
Blacklisting named imports have been added in [v5.12](https://github.com/palantir/tslint/releases/tag/5.12.0)

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
